src/homedelivery/export.js

```diff
@@ -0,0 +1,158 @@
+const HOLIDAYS_QUERY_NAME = 'HolidaySuspensions'
+const SUBSCRIPTIONS_QUERY_NAME = 'Subscriptions'
+
+function getDownloadStream (results: Array<result>, stage: string, queryName: string): Promise<ReadStream> {
+  function getFileName (queryName) {
+    function isTargetQuery (result) {
+      return result.queryName === queryName
+    }
+
+    let filtered = results.filter(isTargetQuery)
+
+    if (filtered.length !== 1) {
+      return null // not sure if there are options in js
+    } else {
+      return filtered[0].fileName
+    }
+  }
+
+  return new Promise((resolve, reject) => {
+    console.log(`getting results file for query: ${queryName}`)
+    let fileName = getFileName(queryName)
+    if (!fileName) {
+      reject(new Error(`Invalid input cannot find unique query called ${queryName}`))
+      return
+    }
+    let path = `${stage}/zuoraExport/${fileName}`
+    resolve(createReadStream(path))
+  })
+}
+
+function getHolidaySuspensions (downloadStream: ReadStream): Promise<Set<string>> {
+  return new Promise((resolve, reject) => {
+    let suspendedSubs = new Set()
+
+    let csvStream = csv.parse({
+      headers: true
+    })
+      .on('data', function (data) {
+        let subName = data['Subscription.Name']
+        suspendedSubs.add(subName)
+      })
+      .on('end', function () {
+        resolve(suspendedSubs)
+      })
+
+    downloadStream.on('error', function (err) {
+      reject(new Error(`error reading holidaySuspensions: ${err}`))
+    })
+      .pipe(csvStream)
+  })
+}
+async function processSubs (downloadStream: ReadStream, deliveryDate: moment, stage: string, holidaySuspensions: Set<string>): Promise<string> {
+  let sentDate = moment().format('DD/MM/YYYY')
+  let chargeDay = deliveryDate.format('dddd')
+  let formattedDeliveryDate = deliveryDate.format('DD/MM/YYYY')
+
+  console.log('loaded ' + holidaySuspensions.size + ' holiday suspensions')
+  let writeCSVStream = csv.createWriteStream({
+    headers: outputHeaders
+  })
+
+  let csvStream = csv.parse({
+    headers: true
+  })
+      .on('data-invalid', function (data) {
+        // TODO CAN WE LOG PII?
+        console.log('ignoring invalid data: ' + data)
+      })
+      .on('data', function (data) {
+        let subscriptionName = data[SUBSCRIPTION_NAME]
+        if (!holidaySuspensions.has(subscriptionName)) {
+          let outputCsvRow = {}
+          outputCsvRow[CUSTOMER_REFERENCE] = subscriptionName
+          outputCsvRow[CUSTOMER_TOWN] = data[CITY]
+          outputCsvRow[CUSTOMER_POSTCODE] = formatPostCode(data[POSTAL_CODE])
+          outputCsvRow[CUSTOMER_ADDRESS_LINE_1] = data[ADDRESS_1]
+          outputCsvRow[CUSTOMER_ADDRESS_LINE_2] = data[ADDRESS_2]
+          outputCsvRow[CUSTOMER_FULL_NAME] = data[FIRST_NAME] + ' ' + data[LAST_NAME]
```

Comment:
Down from SF to Zuora, yes we're working on that. We've stopped syncing up to SF from Zuora which didn't copy this field.